src/server/pegasus_server_impl.cpp

```diff
@@ -3007,6 +3013,97 @@ void pegasus_server_impl::reset_usage_scenario_options(
     target_opts->max_write_buffer_number = base_opts.max_write_buffer_number;
 }
 
+void pegasus_server_impl::recalculate_data_cf_options(
+    const rocksdb::ColumnFamilyOptions &cur_data_cf_opts)
+{
+#define UPDATE_OPTION_IF_NEEDED(option, value, str_value)                                          \
+    if ((value) != cur_data_cf_opts.option) {                                                      \
+        new_options[#option] = (str_value);                                                        \
+    }
```

Comment:
How about:
```
#define UPDATE_OPTION_IF_NEEDED(option, new_value)             \
    if (_data_cf_opts.option != cur_data_cf_opts.option) {     \
        new_options[#option] = #new_value;                     \
    }
```

Reply:
`if (_data_cf_opts.option != cur_data_cf_opts.option) { ` is wrong
use level0_file_num_compaction_trigger as an example: 
1. start_onebox, do not change the config template
```
cur_data_cf_opts.level0_file_num_compaction_trigger = 4
_data_cf_opts.level0_file_num_compaction_trigger = 4
```
2. set `usage scenario`is `bulk load`
```
cur_data_cf_opts.level0_file_num_compaction_trigger = 1000000000
_data_cf_opts.level0_file_num_compaction_trigger = 4
```
3. restart the db, will call LoadLatestOptions() 
```
cur_data_cf_opts.level0_file_num_compaction_trigger = 1000000000
_data_cf_opts.level0_file_num_compaction_trigger = 4
```
At this time `_data_cf_opts.option != cur_data_cf_opts.option`, but in fact the value of option is correct, nothing needs to be done, so need `(value) != cur_data_cf_opts.option) `

Reply:
I meant value and str_value are always the same, you can use one instead.

Reply:
when `new_options[#option] = #new_value;`run  the following statement:
```
UPDATE_OPTION_IF_NEEDED(level0_slowdown_writes_trigger,_data_cf_opts.level0_slowdown_writes_trigger)
```
result is:
```
new_options["level0_slowdown_writes_trigger"] = "_data_cf_opts.level0_slowdown_writes_trigger"
```